backbone.js

```diff
@@ -1140,6 +1140,7 @@
         removed.push(model);
         this._removeReference(model, options);
       }
+      if (models.length > 0 && !options.silent) delete options.index;
```

Comment:
It's not guarding on `this.models`, it's guarding on the passed in `arguments[0] : models` (to be removed).  It's an optimization.  If there were no models to delete, then `options.index` was not set in _removeModels.  Therefore, no need to delete `options.index`.